src/xdp/programinspect.c

```diff
@@ -1205,6 +1326,62 @@ XdpInspect(
             }
             break;
 
+        case XDP_MATCH_ICMPV4_ECHO_REPLY_IP_DST:
+            if (!FrameCache.Ip4Cached) {
+                XdpParseFrame(
+                    Frame, FragmentRing, FragmentExtension, FragmentIndex, VirtualAddressExtension,
+                    &FrameCache, &Program->FrameStorage);
+            }
+
+            if (!FrameCache.IpPayloadValid) {
+                break;
+            }
+
+            if (!FrameCache.Icmp4Cached) {
+                XdpParseIcmp4Header(
+                    Frame, FragmentRing, FragmentExtension, FragmentIndex, VirtualAddressExtension,
+                    &FrameCache.IpPayload, &Program->FrameStorage, &FrameCache);
+            }
+
+            if (FrameCache.Icmp4Valid &&
+                FrameCache.Icmpv4Hdr->Type == ICMP4_ECHOREPLY_TYPE &&
+                FrameCache.Icmpv4Hdr->Code == ICMP4_ECHOREPLY_CODE &&
+                IN4_ADDR_EQUAL(
+                    &FrameCache.Ip4Hdr->DestinationAddress,
+                    &Rule->Pattern.IpMask.Address.Ipv4)) {
+                ASSERT(FrameCache.Ip4Valid);
+                Matched = TRUE;
+            }
+            break;
+
+        case XDP_MATCH_ICMPV6_ECHO_REPLY_IP_DST:
+            if (!FrameCache.Ip6Cached) {
+                XdpParseFrame(
+                    Frame, FragmentRing, FragmentExtension, FragmentIndex, VirtualAddressExtension,
+                    &FrameCache, &Program->FrameStorage);
+            }
+
+            if (!FrameCache.IpPayloadValid) {
```

Comment:
I think you need to check `FrameCache.Ip6Valid` here, too, and similar for v4 above.